python_modules/dagster/dagster/_check/builder.py

```diff
@@ -219,8 +219,11 @@ def _name(target: Optional[TypeOrTupleOfTypes]) -> str:
     if target is None:
         return "None"
 
+    if target is NoneType:
+        return "check.NoneType"
+
     if isinstance(target, tuple):
-        return f"({', '.join(tup_type.__name__ if tup_type is not NoneType else 'check.NoneType' for tup_type in target)})"
+        return f"({', '.join(_name(tup_type) for tup_type in target)})"
```

Comment:
Do we need to guard against the values in the tuple being NoneType like the previous implementation did?

Reply:
I hoisted that up to [1]

Reply:
I mean that the new check in [1] checks that target is nonetype, but the old check checks if values in target are nonetype.

Now if `target = tuple(Nonetype, Nonetype)`, this codepath will be hit and we'll call `_name(Nonetype)`.